### PR TITLE
fix(sui): be consistent about sui config directory

### PR DESCRIPTION
**Pull request by amnn**
## Description
When a `--config` is supplied and it is a directory, we treat it as the sui config directory in some but not all cases. This changes makes it so that if `--config` does not look like a YAML file, then it is treated like it should be the config directory, and that location is used everywhere the config directory is expected.

## Test plan

Manually tested running `sui start`, `sui start --network.config ...`:

```sh
$ sui start
$ sui start --force-regenesis
$ sui start --network.config ~/.sui/sui_config
$ sui start --network.config ~/.sui/sui_config/network.yaml
$ sui start --network.config "$CUSTOM_CONFIG"
$ sui start --network.config "$CUSTOM_CONFIG/network.yaml"
```

A custom config directory can be generated by taking a copy of the current config directory and tweaking it:

```sh
$ cp -r ~/.sui/sui_config $CUSTOM_CONFIG
$ rm "$CUSTOM_CONFIG/sui.keystore" "$CUSTOM_CONFIG/sui.aliases"
$ sed -i '' "s:$HOME/.sui/sui_config:$CUSTOM_CONFIG:" "$CUSTOM_CONFIG/network.yaml"
```

You can test that the local network is up and running by querying for its chain identifier:

```sh
$ curl -LX POST  "http://localhost:9000" \
    --header 'Content-Type: application/json' \
    --data '{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "sui_getChainIdentifier",
  "params": []
}' | jq .
```

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [x] CLI: Support fully overriding the config directory during `sui start`.
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

**Review by stefan-mysten — APPROVED** [review]
Thanks for fixing this, though I am a scared of this change. The existing logic there is a mess :p.

File: crates/sui/src/sui_commands.rs [hunk 1]
```diff
@@ -638,41 +638,59 @@ async fn start(
...
+            }
+
+            Some(config) => (config.join(SUI_NETWORK_CONFIG), config),
+
+            None => {
+                let sui_config = sui_config_dir()?;
+                let network_config = sui_config.join(SUI_NETWORK_CONFIG);
+
+                if !network_config.exists() {
+                    genesis(None, None, None, false, epoch_duration_ms, None, false)
```
Comment (stefan-mysten) [inline crates/sui/src/sui_commands.rs:673]: Wouldn't this potentially override other settings in the `sui_config` dir?

Comment (stefan-mysten) [reply crates/sui/src/sui_commands.rs:673]: Ah, `genesis` will fail if the folder is not empty, so we're probably safe.

Comment (amnn) [reply crates/sui/src/sui_commands.rs:673]: This logic is pre-existing, but judging by the error message, genesis will not run if the directory is non-empty.

**Comment by amnn** [comment]
I'm a little scared as well -- this change actually started as a pretty benign formatting change for the extra long-line containing the error message, but then I couldn't resist pulling the thread that I noticed with the config directories being set in various places...

File: crates/sui/src/sui_commands.rs (hunk 1 above)
Comment (stefan-mysten) [reply crates/sui/src/sui_commands.rs:673]: Yes, I did that messy logic 😅  trying to preserve compatibility between `sui-test-validator` and `sui start`. :D

**Pull request merged**